apps/site/assets/ts/schedule/components/LineDiagram.tsx

```diff
@@ -97,120 +128,196 @@ const LineDiagram = ({
     modalOpen: false
   });
 
+  const isBranchEnd = (index: number): boolean =>
+    index === lineDiagram.length - 1 ||
+    index ===
+      lineDiagram.lastIndexOf(
+        lineDiagram.filter(isOnSecondaryBranch)[
+          lineDiagram.filter(isOnSecondaryBranch).length - 1
+        ]
+      ) ||
+    (index > 0 &&
+      lineDiagram
+        .filter(l => isTerminusStop(l) && !isBeginningStop(l))
+        .includes(lineDiagram[index]));
```

Comment:
These conditionals are a bit tricky to follow — I'm not even sure what that middle part with the `filter`s does. Could we assign each part of the condition to a descriptively-named variable? Or extract functions?

Reply:
Ended up making this logic obsolete in my latest commit.